### PR TITLE
Docs: update mysql docs to reflect editor changes

### LINKED ISSUE #54079: Update documentation for MySQL datasource after migration to the new querybuilder

### PR DESCRIPTION
**Pull request by oscarkilhed**
**What this PR does / why we need it**:
This updates the mysql docs to reflect the changes to the code editor and visual query builder.

**Which issue(s) this PR fixes**:

Fixes #54079

### CODE REVIEW

File: docs/sources/datasources/mysql.md [hunk 1]
```diff
@@ -73,60 +73,51 @@ Example:
...
+### Format
 
-The query editor has a link named `Generated SQL` that shows up after a query has been executed, while in panel edit mode. Click on it and it will expand and show the raw interpolated SQL string that was executed.
-
-### Select table, time column and metric column (FROM)
-
-When you enter edit mode for the first time or add a new query Grafana will try to prefill the query builder with the first table that has a timestamp column and a numeric column.
-
-In the FROM field, Grafana will suggest tables that are in the configured database. To select a table or view in another database that your database user has access to you can manually enter a fully qualified name (database.table) like `otherDb.metrics`.
+The response from MySQL can be formated as either a table or as a time series. To use the time series format one of the columns must be named `time`.
```
Comment (brendamuir) [inline docs/sources/datasources/mysql.md:84]:
```suggestion
The response from MySQL can be formatted as either a table or as a time series. To use the time series format one of the columns must be named `time`.
```

File: docs/sources/datasources/mysql.md [hunk 2]
```diff
@@ -73,60 +73,51 @@ Example:
...
 
-The Time column field refers to the name of the column holding your time values. Selecting a value for the Metric column field is optional. If a value is selected, the Metric column field will be used as the series name.
+### Dataset and Table selection
 
-The metric column suggestions will only contain columns with a text datatype (text, tinytext, mediumtext, longtext, varchar, char).
-If you want to use a column with a different datatype as metric column you may enter the column name with a cast: `CAST(numericColumn as CHAR)`.
-You may also enter arbitrary SQL expressions in the metric column field that evaluate to a text datatype like
-`CONCAT(column1, " ", CAST(numericColumn as CHAR))`.
+In the dataset dropdown you can chose the MySQL database to query. The dropdown will be populated with the databases that the user has access to.
+When the dataset is selected the table dropdown is populated with the tables that are available.
```
Comment (brendamuir) [inline docs/sources/datasources/mysql.md:89]:
```suggestion
In the dataset dropdown, choose the MySQL database to query. The dropdown is be populated with the databases that the user has access to.
When the dataset is selected, the table dropdown is populated with the tables that are available.
```

File: docs/sources/datasources/mysql.md [hunk 3]
```diff
@@ -73,60 +73,51 @@ Example:
...
 
 ### Columns and Aggregation functions (SELECT)
 
-In the `SELECT` row you can specify what columns and functions you want to use.
-In the column field you may write arbitrary expressions instead of a column name like `column1 * column2 / column3`.
+Using the dropdown you can select a column to include inte the data. You can also specify a an optional aggregation function.
```
Comment (brendamuir) [inline docs/sources/datasources/mysql.md:93]:
```suggestion
Using the dropdown,  select a column to include in the data. You can also specify an optional aggregation function.
```

File: docs/sources/datasources/mysql.md [hunk 4]
```diff
@@ -73,60 +73,51 @@ Example:
...
 
-If you use aggregate functions you need to group your resultset. The editor will automatically add a `GROUP BY time` if you add an aggregate function.
-
-You may add further value columns by clicking the plus button and selecting `Column` from the menu. Multiple value columns will be plotted as separate series in the graph panel.
+You may add further value columns by clicking the plus button and another column dropdown will appear.
```
Comment (brendamuir) [inline docs/sources/datasources/mysql.md:95]:
```suggestion
Add further value columns by clicking the plus button and another column dropdown appears.
```

File: docs/sources/datasources/mysql.md [hunk 5]
```diff
@@ -73,60 +73,51 @@ Example:
...
 
 ### Filter data (WHERE)
 
-To add a filter click the plus icon to the right of the `WHERE` condition. You can remove filters by clicking on
-the filter and selecting `Remove`. A filter for the current selected timerange is automatically added to new queries.
+To add a filter flip the switch at the top of the editor.
+Using the first dropdown you can select if all the filters need to match (AND) or if only one of the filters need to match (OR).
+
+To add more columns to filter on use the plus button.
```
Comment (brendamuir) [inline docs/sources/datasources/mysql.md:102]:
```suggestion
To add a filter, flip the switch at the top of the editor.
Using the first dropdown, select if all the filters need to match (AND) or if only one of the filters needs to match (OR).

To add more columns to filter on use the plus button.
```

File: docs/sources/datasources/mysql.md [hunk 6]
```diff
@@ -73,60 +73,51 @@ Example:
...
 
 ### Group By
 
-To group by time or any other columns click the plus icon at the end of the GROUP BY row. The suggestion dropdown will only show text columns of your currently selected table but you may manually enter any column.
-You can remove the group by clicking on the item and then selecting `Remove`.
+To group the results by a column, flip the group switch at the top of the editor. You can then choose what column to group the results by. The group by clause can be removed by pressing the X button
+
```
Comment (brendamuir) [inline docs/sources/datasources/mysql.md:107]:
```suggestion
To group the results by column, flip the group switch at the top of the editor. You can then choose which column to group the results by. The group by clause can be removed by pressing the X button.

```

File: docs/sources/datasources/mysql.md [hunk 7]
```diff
@@ -73,60 +73,51 @@ Example:
...
+### Preview
 
-If you add any grouping, all selected columns need to have an aggregate function applied. The query builder will automatically add aggregate functions to all columns without aggregate functions when you add groupings.
+By flipping the preview switch at the top of the editor you can get a preview of the SQL query generated by the query builder.
```
Comment (brendamuir) [inline docs/sources/datasources/mysql.md:110]:
```suggestion
By flipping the preview switch at the top of the editor, you can get a preview of the SQL query generated by the query builder.
```

File: docs/sources/datasources/mysql.md [hunk 8]
```diff
@@ -73,60 +73,51 @@ Example:
...
 
-#### Gap Filling
+## Code editor
 
-Grafana can fill in missing values when you group by time. The time function accepts two arguments. The first argument is the time window that you would like to group by, and the second argument is the value you want Grafana to fill missing items with.
+{{< figure src="/static/img/docs/v92/mysql_code_editor.png" class="docs-image--no-shadow" }}
 
-### Text Editor Mode (RAW)
+To make advanced queries you can switch to the code editor by clicking `code` in the top right corner of the editor. The code editor support autocompletion of tables, columns, SQL keywords, standard sql functions, grafana template variables and grafana macros. Columns cannot be completed before a table has been specified.
```
Comment (brendamuir) [inline docs/sources/datasources/mysql.md:116]:
```suggestion
To make advanced queries, switch to the code editor by clicking `code` in the top right corner of the editor. The code editor support autocompletion of tables, columns, SQL keywords, standard sql functions, Grafana template variables and Grafana macros. Columns cannot be completed before a table has been specified.
```

File: docs/sources/datasources/mysql.md [hunk 9]
```diff
@@ -73,60 +73,51 @@ Example:
 
 You can use wildcards (`*`) in place of database or table if you want to grant access to more databases and tables.
 
-## Query Editor
+## Query Builder
```
Comment (knylander-grafana) [inline docs/sources/datasources/mysql.md:76]: Suggesting sentence case heading. 
```suggestion
## Query builder
```

File: docs/sources/datasources/mysql.md [hunk 10]
```diff
@@ -73,60 +73,51 @@ Example:
...
 
-> Only available in Grafana v5.4+.
+{{< figure src="/static/img/docs/v92/mysql_query_builder.png" class="docs-image--no-shadow" }}
 
-{{< figure src="/static/img/docs/v54/mysql_query_still.png" class="docs-image--no-shadow" animated-gif="/static/img/docs/v54/mysql_query.gif" >}}
+The MySQL query builder is available when editing a panel using a MySQL datasource. The built query can be run by pressing the `Run query` button in the top right corner of the editor.
```
Comment (knylander-grafana) [inline docs/sources/datasources/mysql.md:80]:
```suggestion
The MySQL query builder is available when editing a panel using a MySQL data source. The built query can be run by pressing the `Run query` button in the top right corner of the editor.
```

**Review by gguillotte-grafana — APPROVED** [review]
Approving, but this is still blocked by grafana/website#10149 to add or update the images.

**Pull request merged**